oap-server/server-storage-plugin/storage-banyandb-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/banyandb/BanyanDBIndexInstaller.java

```diff
@@ -240,18 +259,49 @@ private ResourceExist checkResourceExistence(MetadataRegistry.SchemaMetadata met
                                                                                    IntervalRule.Unit.UNIT_DAY)
                                                                                .setNum(
                                                                                    metadata.getTtlDays())));
+        BanyandbCommon.ResourceOpts.Builder optsBuilder = BanyandbCommon.ResourceOpts.newBuilder().setShardNum(metadata.getShard());
+
         switch (metadata.getKind()) {
             case STREAM:
+                optsBuilder.setSegmentInterval(
+                IntervalRule.newBuilder()
+                    .setUnit(
+                        IntervalRule.Unit.UNIT_DAY)
+                    .setNum(
+                        metadata.getSegmentIntervalDays()))
+                .setTtl(
+                    IntervalRule.newBuilder()
+                        .setUnit(
+                            IntervalRule.Unit.UNIT_DAY)
+                        .setNum(
+                            metadata.getTtlDays()));
                 resourceExist = client.existStream(metadata.getGroup(), metadata.name());
                 gBuilder.setCatalog(BanyandbCommon.Catalog.CATALOG_STREAM).build();
                 break;
             case MEASURE:
+                optsBuilder.setSegmentInterval(
+                        IntervalRule.newBuilder()
+                            .setUnit(
+                                IntervalRule.Unit.UNIT_DAY)
+                            .setNum(
+                                metadata.getSegmentIntervalDays()))
+                    .setTtl(
+                        IntervalRule.newBuilder()
+                            .setUnit(
+                                IntervalRule.Unit.UNIT_DAY)
+                            .setNum(
+                                metadata.getTtlDays()));
                 resourceExist = client.existMeasure(metadata.getGroup(), metadata.name());
                 gBuilder.setCatalog(BanyandbCommon.Catalog.CATALOG_MEASURE).build();
                 break;
+            case Property:
+                resourceExist = client.existProperty(metadata.getGroup(), metadata.name());
+                gBuilder.setCatalog(BanyandbCommon.Catalog.CATALOG_PROPERTY).build();
+                break;
             default:
                 throw new IllegalStateException("unknown metadata kind: " + metadata.getKind());
         }
+        gBuilder.setResourceOpts(optsBuilder.build());
```

Comment:
We need to make the logic clear
1. TTL is not affecting the property, so, let's move the initialization logic into `case`.
2. Is shard of property available? If so, let's move it into property case.

Reply:
> 1. TTL is not affecting the property, so, let's move the initialization logic into `case`.

No, we can't do that. The "shard" is also a part of ResourceOpts. Let's keep the setResourceOpts at a global level.